Frameworks/Foundation/NSDictionary.mm

```diff
@@ -793,53 +769,94 @@ - (NSArray*)objectsForKeys:(NSArray*)keys notFoundMarker:(id)notFoundMarker {
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (NSSet*)keysOfEntriesPassingTest:(BOOL (^)(id, id, BOOL*))predicate {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self keysOfEntriesWithOptions:0 passingTest:predicate];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-- (NSSet*)keysOfEntriesWithOptions:(NSEnumerationOptions)opts passingTest:(BOOL (^)(id, id, BOOL*))predicate {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSSet*)keysOfEntriesWithOptions:(NSEnumerationOptions)options passingTest:(BOOL (^)(id, id, BOOL*))predicate {
+    __block NSMutableSet* ret = [NSMutableSet setWithCapacity:[self count]];
+    [self enumerateKeysAndObjectsWithOptions:options
+                                  usingBlock:^(id key, id obj, BOOL* stop) {
+                                      if (predicate(key, obj, stop)) {
+                                          [ret addObject:key];
+                                      }
+                                  }];
+
+    return ret;
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Caveat
+ @Notes atomically parameter not supported
 */
-- (BOOL)writeToURL:(NSURL*)aURL atomically:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (BOOL)writeToURL:(NSURL*)url atomically:(BOOL)atomically {
+    NSError* error = nil;
+    NSData* data = [NSPropertyListSerialization dataWithPropertyList:self format:NSPropertyListXMLFormat_v1_0 options:0 error:&error];
+    if (error != nil) {
+        return NO;
+    }
+
+    return [data writeToURL:url atomically:atomically];
 }
 
 /**
  @Status Stub
  @Notes
 */
 - (NSString*)descriptionWithLocale:(id)locale {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self descriptionWithLocale:locale indent:0];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-- (NSString*)descriptionWithLocale:(id)locale indent:(NSUInteger)level {
-    UNIMPLEMENTED();
-    return StubReturn();
+- (NSString*)descriptionWithLocale:(id)locale indent:(NSUInteger)indent {
+    NSMutableString* s = [NSMutableString new];
+    for (NSUInteger i = 0; i < indent; ++i) {
+        [s appendString:@"    "];
+    }
```

Comment:
this would be good only if we had it: [https://developer.apple.com/reference/foundation/nsstring/1416395-stringbypaddingtolength](https://developer.apple.com/reference/foundation/nsstring/1416395-stringbypaddingtolength)

but you can do this:
  NSMutableString* s = [NSMutableString stringWithFormat:@"%10c"," "];